libs/ha/ha_common_libs_k8s.py

```diff
@@ -1846,3 +1862,172 @@ def dnld_obj_verify_chcksm(self, s3_test_obj, bucket, obj, download_path, upload
             return False, download_checksum
         LOGGER.info("Successfully downloaded the object and verified the checksum")
         return True, download_checksum
+
+    @staticmethod
+    def parallel_put_object(event, s3_test_obj, bkt_name: str, obj_name: str, output=None,
+                            **kwargs):
+        """
+        Function to upload object
+        Upload an object to a versioning enabled/suspended bucket and return list of dictionary of
+        uploaded versions in parallel or non parallel.
+        :param event: event to intimate thread about main thread operations
+        :param s3_test_obj: Object of the s3 test lib
+        :param bkt_name: Bucket name for calling PUT Object
+        :param obj_name: Object name for calling PUT Object
+        :param output: Output queue in which results should be put
+        :keyword file_path: File path that can be used for PUT Object call
+        :keyword count: Count for number of PUT object call
+        :keyword chk_null_version: True, if 'null' version id is expected, else False
+        :keyword background: Set to true if background function call
+        :keyword is_unversioned: Set to true if object is uploaded to an unversioned bucket
+        Can be used for setting up pre-existing objects before enabling/suspending bucket
+        versioning
+        :return: Tuple (bool, list)
+        """
+        chk_null_version = kwargs.get("chk_null_version", False)
+        file_path = kwargs.get("file_path")
```

Comment:
If file path is not given by user, provide some default path here.

Reply:
> If file path is not given by user, provide some default path here.

Since, we are putting same object value with same etag, throughout the test we have to use same file. That's why I have not given file path, as we are calling the same function for many put request.